File: fieldmate/src/main/java/com/nguyencong/fieldmate/service/AuthService.java
package com.nguyencong.fieldmate.service;

import java.io.IOException;

import com.nguyencong.fieldmate.dto.request.LoginRequest;
import com.nguyencong.fieldmate.dto.request.RegisterRequest;
import com.nguyencong.fieldmate.dto.response.AuthResponse;
import com.nguyencong.fieldmate.dto.response.UserResponse;

public interface AuthService {
    UserResponse registerUser(RegisterRequest request) throws IOException;
    AuthResponse login(LoginRequest request);
}

File: front-end/fieldmate-web/src/services/auth.service.ts
import {
  fieldmateClient,
  fieldmateEndpoints,
} from "@/configs/fieldmate-client";
import type {
  AuthResponse,
  LoginRequest,
  RegisterRequest,
  UserResponse,
} from "@/types/auth";

export const authService = {
  async login(request: LoginRequest) {
    const response = await fieldmateClient.post<AuthResponse>(
      fieldmateEndpoints.login,
      request,
    );

    return response.data;
  },

  async register(request: RegisterRequest) {
    const formData = new FormData();

    formData.append("email", request.email.trim());
    formData.append("password", request.password);
    formData.append("phoneNumber", request.phoneNumber.trim());
    formData.append("firstName", request.firstName.trim());
    formData.append("lastName", request.lastName.trim());
    formData.append("role", request.role);

    if (request.avatar) {
      formData.append("avatar", request.avatar);
    }

    const response = await fieldmateClient.post<UserResponse>(
      fieldmateEndpoints.register,
      formData,
      {
        headers: {
          "Content-Type": "multipart/form-data",
        },
      },
    );

    return response.data;
  },
};
